pylint/checkers/refactoring.py

```diff
@@ -475,6 +483,51 @@ def visit_boolop(self, node):
                              node=node,
                              args=(duplicated_name, ', '.join(names)))
 
+    def _check_consider_using_in(self, node):
+        allowed_ops = {'or': '==',
+                       'and': '!='}
+
+        if node.op not in allowed_ops or len(node.values) < 2:
+            return
+
+        for value in node.values:
+            if (not isinstance(value, astroid.Compare)
+                    or len(value.ops) != 1
+                    or value.ops[0][0] not in allowed_ops[node.op]):
+                return
+
+        variable_sets = []
```

Comment:
This looks neat. Can we break this function in multiple smaller functions? I see at least the common variables block to be a candidate for a shorter function.

Reply:
I already thought about extracting some functions, as this seems lengthy.
The problem is tough, that if we extract functions, those would be overly specific and i doubt they would add any value to the outer context or even be used a second time.
For example a function for the common variables block would have to be named something like the following in order not to be confusing and not to claim to be too generic, so others might use them in a wrong way and be confused about the result:
```python
common_variables, values = _extract_common_variables_and_values_from_boolop(node)
```
And even that may not be specific enough as the fact that we only consider comparisons in this function is still missing.
I tried to make it more obvious whats going on in this block, but maybe you could outline what is troubling you in detail.

Reply:
Thanks for the reply. The lengthy function was worrying me as it might make the code harder to understand. I'm not worried about the generality of the function as you said I don't imagine this being reused. As a middle ground what would you say about either adding some comments or making the common variables block an inner function to this method?

Reply:
Sure, I'd love to make the code easier to understand.
In the most recent commit I've reworded some things and changed the comments a bit.
Did it help enough?
Please be honest if not :smile: